### PR TITLE
feat(ui): add Lighthouse request outcome feedback

### PR DESCRIPTION
**Pull request by Alan-TheGentleman**
### Context

Adds stateless Lighthouse request-outcome feedback directly from the UI, following the existing PostHog-backed `Prowler Feedback` widget pattern.

No public issue is linked for this internal product integration. No backend endpoint or persistence layer is required.

### Description

- Loads the launched PostHog API survey named `Lighthouse Request Outcome Feedback` once per Lighthouse chat view.
- Fails closed when the survey is missing or its three-question schema is incompatible, including optionality, question types, choices, or question identities.
- Associates feedback controls with the adjacent persisted user message; assistant messages and optimistic unpersisted messages are never feedback targets.
- Thumbs-up immediately captures rating-only feedback.
- Thumbs-down opens a popover with six optional reasons and optional free-text details.
- Captures `survey sent` events through the existing `posthog-js` singleton without a server action or API request.
- Uses one submission ID per logical response, maps ratings to the PostHog two-emoji scale, and correlates the persisted user-message ID through `$ai_trace_id`.
- Emits cumulative negative-response snapshots so PostHog response collapsing cannot discard previously selected reasons.
- Keeps details trimmed and bounded to 2,000 characters and normalizes reason values before capture.
- Preserves feedback availability after terminal response errors.

Deployment requirement: each environment must expose the launched API survey with the expected rating, optional multiple-choice reasons, and optional open-details questions. No survey or production activation is performed by this PR.

### Steps to review

1. Confirm the chat view subscribes to surveys once and releases the subscription on unmount.
2. Confirm feedback controls appear only when the exact API survey schema is available and the initiating user message is persisted.
3. Confirm thumbs-up captures one rating-only `survey sent` event with the persisted user-message ID as `$ai_trace_id`.
4. Confirm thumbs-down supports rating-only, reasons-only, details-only, or combined submissions.
5. Confirm negative events share one `$survey_submission_id` and carry cumulative response properties.
6. Confirm no Lighthouse feedback server action, adapter request, endpoint path, or backend dependency remains.
7. Review the focused survey-builder, chat-page, message-bubble, action, and adapter tests.

### Checklist

<details>

<summary><b>Community Checklist</b></summary>

- [ ] This feature/issue is listed in the open issues or roadmap.prowler.com
- [ ] Is it assigned to me
- [x] I have reviewed the open pull requests and confirmed there is no existing PR that implements the same outcome

</details>

- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following the project specification.
- [x] Review if backport is needed: no backport required.
- [x] Review if the README needs changes: no README change required.

#### SDK/CLI

- Are there new checks included in this PR? No

#### UI

- [x] All task requirements work as expected on the UI
- [x] No npm dependencies were added or updated
- [ ] Screenshots/Video - Mobile: not included
- [ ] Screenshots/Video - Tablet: not included
- [ ] Screenshots/Video - Desktop: not included
- [x] A changelog fragment is included under `ui/changelog.d/`

#### API

- Not applicable. Lighthouse feedback is captured directly through the existing frontend PostHog client.

#### MCP Server

- Not applicable.

### Verification

- 87 focused UI tests passed.
- TypeScript typecheck passed.
- ESLint and Prettier checks passed.
- Commit hooks passed the full UI unit-test gate.
- Dead-code searches found no Lighthouse feedback API action, adapter request, or endpoint path.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

## Summary by CodeRabbit

* **New Features**
  * Added thumbs-up and thumbs-down feedback controls for Lighthouse assistant responses.
  * Negative feedback supports selectable reasons and optional details.
  * Added validation, submission states, retry, cancellation, and dismissal behaviors.
  * Feedback is available for eligible responses, including those affected by errors.
  * Added reusable feedback form support for survey-based feedback.

* **Bug Fixes**
  * Preserved feedback availability after terminal response-stream errors.

* **Tests**
  * Added comprehensive coverage for feedback visibility, validation, submission, surveys, and edge cases.

### CODE REVIEW

File: ui/app/(prowler)/lighthouse/_components/chat/lighthouse-v2-chat-view.tsx [hunk 1]
```diff
@@ -200,14 +207,22 @@ export function LighthouseV2ChatView({
...
             {messages.map((message, index) => {
-              const skillRun = getSkillRunFromLaunch(
-                message,
-                messages[index - 1],
-              );
+              const previousMessage = messages[index - 1];
+              const skillRun = getSkillRunFromLaunch(message, previousMessage);
+              // The assistant owns the controls visually; its adjacent persisted
+              // user turn remains the API task/trace feedback target.
+              const feedbackTarget =
```
Comment (alejandrobailo) [inline ui/app/(prowler)/lighthouse/_components/chat/lighthouse-v2-chat-view.tsx:221]: Could we keep the feedback controls available when the run ends with a terminal error? Right now, feedbackTarget is only set for assistant messages, but the terminal-error path refreshes to the persisted user message only, so there is no assistant bubble where the controls can render. I reproduced this by emitting the terminal error: the error banner appears, but neither rating button does. Please add a UI state for the failed outcome, keep the persisted user-message ID as the trace ID, and cover this flow in lighthouse-v2-chat-page.test.tsx.

**Review by alejandrobailo — CHANGES_REQUESTED** [review]
Good job overall! Could you please address the inline comment and the three still-valid unresolved threads before merging? Thanks!

File: ui/app/(prowler)/lighthouse/_actions/lighthouse-v2.test.ts [hunk 2]
```diff
@@ -158,6 +159,72 @@ describe("Lighthouse v2 session write actions", () => {
...
+            type: "lighthouse-message-feedback",
+            attributes: {
+              rating: "down",
+              reasons: ["Low quality", "Other"],
+              details: "Missing evidence",
+            },
+          },
+        }),
+      }),
+    );
```
[the remark this answers is not in the document]
Comment (alejandrobailo) [reply ui/app/(prowler)/lighthouse/_actions/lighthouse-v2.test.ts]: Thanks! This no longer applies after the PostHog refactor in 5294dcd. The feedback Server Action and mutateEmpty request were removed, so there is no feedback JSON:API Content-Type header left to assert.

File: ui/app/(prowler)/lighthouse/_components/chat/message-bubble.tsx [hunk 3]
```diff
@@ -229,6 +271,177 @@ function MessageMeta({
...
+        "Feedback is temporarily unavailable. Please try again.";
+      setError(errorMessage);
+      if (submittedRating === LIGHTHOUSE_V2_FEEDBACK_RATING.UP) {
+        setRating(null);
+        toast({ variant: "destructive", description: errorMessage });
+      }
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (alejandrobailo) [reply ui/app/(prowler)/lighthouse/_components/chat/message-bubble.tsx]: Thanks! This thread targets the removed Server Action flow. The current PostHog implementation intentionally keeps the controls reusable and assigns a fresh survey submission ID to each capture; message-bubble.test.tsx covers that behavior. We are keeping it as designed.

File: ui/app/(prowler)/lighthouse/_types/sessions.ts [hunk 4]
```diff
@@ -37,6 +37,26 @@ export interface LighthouseV2Part {
...
+  STYLE: "Don't like the style",
+  INSTRUCTIONS: "Didn't fully follow instructions",
+  QUALITY: "Low quality",
+  BIAS: "Biased",
+  SAFETY_OR_LEGAL: "Safety or legal concern",
+  OTHER: "Other",
+} as const;
+
+export type LighthouseV2FeedbackReason =
+  (typeof LIGHTHOUSE_V2_FEEDBACK_REASON)[keyof typeof LIGHTHOUSE_V2_FEEDBACK_REASON];
```
[the remark this answers is not in the document]
Comment (alejandrobailo) [reply ui/app/(prowler)/lighthouse/_types/sessions.ts]: Thanks! We should keep the configured PostHog choice strings here. The resolver validates the survey choices against this exact ordered allowlist, and PostHog expects the selected choice text as the response value. Switching only the client to private codes would break the configured survey contract, so I am leaving this unchanged.

File: ui/app/(prowler)/lighthouse/_lib/chat-store.ts [hunk 5]
```diff
@@ -193,7 +210,14 @@ export function createLighthouseChatStore(
...
         if (refreshed) {
-          set({ streamState: createInitialLighthouseV2StreamState() });
+          const failedOutcomeMessageId =
+            event.type === LIGHTHOUSE_V2_SSE_EVENT.ERROR
+              ? findLastPersistedUserMessageId(get().messages)
+              : null;
+          set({
+            streamState: createInitialLighthouseV2StreamState(),
+            failedOutcomeMessageId,
+          });
```
[the remark this answers is not in the document]
Comment (alejandrobailo) [reply ui/app/(prowler)/lighthouse/_lib/chat-store.ts]: Good catch, thanks! Fixed in `47aed5933` by retaining terminal SSE events for the active submission and applying the persisted-message refresh after the pending send settles. I also added regression coverage for both early `error` and `message.end` events.

**Review by alejandrobailo — APPROVED** [review]
